model/task_queue_service.go

```diff
@@ -342,14 +410,16 @@ func (t *basicCachedDispatcherImpl) nextTaskGroupTask(unit schedulableUnit) *Tas
 		t.units[unit.id].tasks[i].IsDispatched = true
 		unit.tasks[i].IsDispatched = true
 
+		// It's running (or already ran) on another host.
 		if nextTaskFromDB.StartTime != util.ZeroTime {
 			continue
 		}
-		// Don't cache dispatched status when returning the next TaskQueueItem - in case the task fails to start.
-		return &nextTask
+		// Don't cache dispatched status when returning the nextTaskQueueItem - in case the task fails to start.
```

Comment:
I don't remember what this means. It seems like maybe we should cache the dispatched status, which we do on 410. Does that make sense?

Reply:
Removing this redundant comment.